scripts/cmake-build.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftNIO open source project
+##
+## Copyright (c) 2025 Apple Inc. and the SwiftNIO project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+set -uo pipefail
+
+log() { printf -- "** %s\n" "$*" >&2; }
+error() { printf -- "** ERROR: %s\n" "$*" >&2; }
+fatal() { error "$@"; exit 1; }
+
+version="${CMAKE_VERSION:="3.31.6"}"
+target_dir="${TARGET_DIRECTORY:=""}"
+
+if [ -z "$version" ]; then
+  fatal "CMake version must be specified."
+fi
+if [ -z "$target_dir" ]; then
+  fatal "Target directory must be specified."
+fi
+
+log "Installing tools for this script"
+apt-get update -y -q && apt-get install -y -q curl rsync
```

Comment:
All your `apt-get` commands can just be `apt`.

Reply:
Separately, why do we have installation of tools here _and_ in the GitHub Actions YAML?

Can it be in one place. Is it more portable for it to be in the YAML, where the distro is also declared, since then this script would work on other distros, which use different package managers.

Reply:
I've tried to have installation of tools just before we use them so that the components are modular. I've changed this line to also check if it's already installed with a `which foo` check to try and make it more lightweight.

Reply:
I've switched the new build script to this approach, it agree it's good to have it portable